internal/xcuitest/config.go

```diff
@@ -55,8 +56,44 @@ type Xcuitest struct {
 
 // TestOptions represents the xcuitest test filter options configuration.
 type TestOptions struct {
-	NotClass []string `yaml:"notClass,omitempty" json:"notClass"`
-	Class    []string `yaml:"class,omitempty" json:"class"`
+	NotClass                          []string `yaml:"notClass,omitempty" json:"notClass"`
+	Class                             []string `yaml:"class,omitempty" json:"class"`
+	TestLanguage                      string   `yaml:"testLanguage,omitempty" json:"testLanguage"`
+	TestRegion                        string   `yaml:"testRegion,omitempty" json:"testRegion"`
+	TestTimeoutsEnabled               string   `yaml:"testTimeoutsEnabled,omitempty" json:"testTimeoutsEnabled"`
+	MaximumTestExecutionTimeAllowance int      `yaml:"maximumTestExecutionTimeAllowance,omitempty" json:"maximumTestExecutionTimeAllowance"`
+	DefaultTestExecutionTimeAllowance int      `yaml:"defaultTestExecutionTimeAllowance,omitempty" json:"defaultTestExecutionTimeAllowance"`
+}
+
+// ToMap converts the TestOptions to a map where the keys are derived from json struct tags.
+func (t TestOptions) ToMap() map[string]interface{} {
```

Comment:
Since it's a toMap function, is it better to use json.Marshal() and json.Unmarshal() ?

Reply:
Like this example: https://go.dev/play/p/Pba6_YyZ9Ws

Although int is still int, not a string. 🤔

Reply:
`Marshal` returns a `[]byte`. `job.StartOptions` specifically wants a `map[string]interface{}` for `TestOptions`.

Reply:
But you can call `json.Unmarshal()` to unmarshal `[]byte` to `map[string]interface{}`

Reply:
There's some type casting happening though in the current implementation.